server/controllers/users.js

```diff
@@ -0,0 +1,198 @@
+import jwt from 'jsonwebtoken';
+import db from '../models';
+import Helpers from './helpers';
+
+/**
+ * Secret token for jsonwebtoken
+ */
+const secret = process.env.SECRET || 'my secret key';
+
+const usersController = {
+  /**
+   * Login a user
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} Response object
+   */
+  login(req, res) {
```

Comment:
This login is fine as it is but it has just one little flaw. It can be hacked with what is called the time attack. Here, you are trying to find the user and if the user is not found, you return an error. This returns faster than when a user is found and you try to decode the password. A hacker will know when he has guessed an email correctly and when he has not. A work around this is to create a new instance of a user without a password if the user is not found and ALWAYS check for the password. In this case, it will take about the same time for when a correct user is found or not.
```JavaScript
db.User.findOne({ where: { email: req.body.email } })
      .then((user) => {
         const user = user || // make a new user instance but don't save it to the database.
           if (user.validatePassword(req.body.password)) {
             const token = jwt.sign({ userId: user.id, roleId: user.roleId}, 
             secret, { expiresIn: '1 day' });
             return res.status(200).send({token, user});
           }else{
               res.status(401)
              .send({ message: 'Invalid Login Details!' });
         }
       });
   },
 ```
This might not be totally correct, but I think you should get the idea.